src/main/java/org/opensearch/sdk/SDKClient.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.io.IOException;
+
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestClientBuilder;
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch.indices.CreateIndexRequest;
+import org.opensearch.client.opensearch.indices.CreateIndexResponse;
+import org.opensearch.client.transport.OpenSearchTransport;
+import org.opensearch.client.transport.rest_client.RestClientTransport;
+
+/**
+ * This class creates a Client for SDK to make requests to OpenSearch
+ */
+public class SDKClient {
+    private final Logger logger = LogManager.getLogger(SDKClient.class);
+    private OpenSearchClient client;
+
+    /**
+     * Creates client for SDK
+     * @throws IOException if client failed
+     */
+    public void createClient() throws IOException {
+        String endpoint = "localhost";
+        String username = "admin";
+        String password = "admin";
+        String protocol = "http";
+        int port = 9200;
```

Comment:
Open to suggestion for a better way to store these attributes.

Reply:
I think the right way to do this is when OpenSearch sends an ExtensionsInitRequest, it should pass this information instead of hard coding in our configuration files. It makes it simple and easy to use. 

Using the configuration file, we have to configure OpenSearch with `extensions.yml`, and again configure a lot of params for the extension. We should be relying on transport as much as possible since the init communication happens.

Reply:
Created https://github.com/opensearch-project/OpenSearch/pull/4197 to address the hostAddress and Port in the `ExtensionsInitRequest`.